### ISSUE TITLE
Question / possible config issue: cockpit-tls: $RUNTIME_DIRECTORY environment variable must be set to a private directory

### ISSUE THREAD

**Issue by jedvod** [cockpit-project/cockpit#15284]
```
# /usr/lib/systemd/system/cockpit.service
[Unit]
Description=Cockpit Web Service
Documentation=man:cockpit-ws(8)
Requires=cockpit.socket
Requires=cockpit-wsinstance-http.socket cockpit-wsinstance-http-redirect.socket cockpit-wsinstance-https-factory.socket
After=cockpit-wsinstance-http.socket cockpit-wsinstance-http-redirect.socket cockpit-wsinstance-https-factory.socket

[Service]
RuntimeDirectory=cockpit/tls
# systemd ≥ 241 sets this automatically
Environment=RUNTIME_DIRECTORY=/run/cockpit/tls
ExecStartPre=+/usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-ws --selinux-type=etc_t
ExecStart=/usr/libexec/cockpit-tls
User=cockpit-ws
Group=cockpit-ws
NoNewPrivileges=true
ProtectSystem=strict
ProtectHome=true
PrivateTmp=true
PrivateDevices=true
ProtectKernelTunables=true
RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
MemoryDenyWriteExecute=true
```

error:
```
cockpit-tls: $RUNTIME_DIRECTORY environment variable must be set to a private directory
```

**Comment by jedvod** [comment]
centos8-stream

**Comment by martinpitt** [comment]
Curious, after #15248 this is the second report of such a thing within a week -- (but the latter was on Ubuntu and was a local mis-installation).

Please describe exactly what you did, any possible customization of your system, and give the output of `sudo systemctl status cockpit.socket cockpit.service`. Please also start

   sudo journalctl -f

then open the  cockpit page, get the error, and then copy&paste the entire log output. Thanks!

(I.e. this is not a general error on CentOS 8 stream -- we continuously test that)

**Comment by jedvod** [comment]
nevermind. Issue with my certificates.

After updating them, I tried to use my full CA bundle, which didn't work, so I used extracted part of it and used it in a certificate formed for https specifically, and then added the key. 

When testing, I opened the .service file and extracted the environment variable so I could also set it in the command line, and then tried to run the ExecStart command, but the error popped up for the runtime directory, which puzzled me as I hadn't changed anything but my certificates.

The `journalctl -u cockpit` only showed start and stop with failures.. no details, when viewing via `journalctl -f` like you suggested, I found the issue was pertaining to the key not matching the certificate, (I was still using the CA bundle at the time).

Apologies for any inconvenience.

Thanks
